### ISSUE TITLE
intermittent failure in sparse test

### ISSUE THREAD

**Issue by JeffBezanson** [JuliaLang/julia#24076]
E.g. here: https://travis-ci.org/JuliaLang/julia/jobs/285862539

```
Expression: !(Base.USE_GPL_LIBS) || abs(det(factorize(Symmetric(A)))) ≈ abs(det(factorize(Array(A))))
  MethodError: no method matching logabsdet(::Base.LinAlg.LDLt{Float64,SymTridiagonal{Float64,Array{Float64,1}}})
  Closest candidates are:
    logabsdet(!Matched::Base.LinAlg.UnitUpperTriangular{T,S} where S<:(AbstractArray{T,2} where T)) where T at linalg/triangular.jl:2193
    logabsdet(!Matched::Base.LinAlg.UnitLowerTriangular{T,S} where S<:(AbstractArray{T,2} where T)) where T at linalg/triangular.jl:2194
    logabsdet(!Matched::Union{LowerTriangular{T,S} where S<:(AbstractArray{T,2} where T), UpperTriangular{T,S} where S<:(AbstractArray{T,2} where T)}) where T at linalg/triangular.jl:2198
```

**Comment by JeffBezanson** [comment]
https://travis-ci.org/JuliaLang/julia/jobs/285889609

**Comment by JeffBezanson** [comment]
https://ci.appveyor.com/project/JuliaLang/julia/build/1.0.19831/job/xk3spyg0gavfrmvr

**Comment by rfourquet** [comment]
@vtjnash may have found a cause [here](https://github.com/JuliaLang/julia/pull/16940#issuecomment-335251951), see also my answer to it.

**Comment by vtjnash** [comment]
`../julia runtests.jl sparse/sparse --seed=0xa57677a41a6158abdaf4fc54c55c744c`
reproduces it (thanks @rfourquet for the hint)

**Comment by vtjnash** [comment]
Also linalg/lq: https://travis-ci.org/JuliaLang/julia/jobs/285885653

**Comment by andreasnoack** [comment]
These shouldn't be hard to fix. We'll need to set more seeds explicitly which is fine. I'll open a PR soon. However, the new seed actually revealed a missing method. The reason for this is that `factorize` is so type unstable. The new random matrix happens to be symmetric tridiagonal whereas the previous was just symmetric or pd.

**Comment by andreasnoack** [comment]
However, the LU failure is more problematic. See https://github.com/JuliaLang/julia/blob/10d470d93203e39fc5bae96333892af7ff4ce202/test/linalg/lu.jl#L196-L208. There shouldn't be anything random in that test so something more serious is going on there.

**Comment by andreasnoack** [comment]
It looks like `Hinv` occasionally has a single element set to zero instead the right value (the (36,42) element when I was able to reproduce locally). Notice again that this should be completely deterministic. There are no random numbers here.

**Comment by kshyatt** [comment]
That line, with all its wrapping, is making me really sad :(

**Comment by andreasnoack** [comment]
I think the issue is somehow related to the comprehension. If I change the comprehension for to an array allocation and a double loop then I'm unable to get the corruption but with the comprehension it happens quite frequently. Hence, we could probably make the tests pass by avoiding the comprehension but it would probably be good to understand why this can happen at all.

**Comment by andreasnoack** [comment]
Fixed by #24085. The other issue is genuine and tracked in #24098.
